src/main/java/com/google/sps/servlets/IPGeolocationServlet.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps.servlets;
+
+import com.google.sps.dao.AnalysisDao;
+import com.google.sps.dao.GeolocationDao;
+import com.google.sps.models.PCAPAnalysis;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/ipgeolocation")
+public class IPGeolocationServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String id = getParameter(request, "PCAPId", "");
```

Comment:
You could create a default PCAPID and store it as a constant. Your getParameter could return that instead. For now add a TODO may b until you add a default pcap id to your data store / mocks.

Reply:
Yep I changed the implementation in the new commits. For now I just set a default mock pcap